tracer/test/Datadog.Trace.Tests/DataStreamsMonitoring/DataStreamsManagerTests.cs

```diff
@@ -209,6 +209,33 @@ public void WhenEnabled_SetCheckpoint_SetsSpanTags()
         span.Tags.GetTag("pathway.hash").Should().NotBeNull();
     }
 
+    [Fact]
+    public async Task WhenEnabled_OneConsumeTwoProduceUsesTwiceConsumePathway()
+    {
+        var dsm = GetDataStreamManager(enabled: true, out var writer);
+
+        dsm.SetCheckpoint(parentPathway: null, CheckpointKind.Consume, new[] { "in" }, payloadSizeBytes: 100, timeInQueueMs: 100);
+
+        dsm.SetCheckpoint(parentPathway: null, CheckpointKind.Produce, new[] { "out" }, payloadSizeBytes: 100, timeInQueueMs: 100);
+        dsm.SetCheckpoint(parentPathway: null, CheckpointKind.Produce, new[] { "out" }, payloadSizeBytes: 100, timeInQueueMs: 100);
+
+        await dsm.DisposeAsync();
+
+        writer.Points.Should().HaveCount(expected: 3);
+        var points = writer.Points.ToArray();
+        // checking that points are in the expected order
+        points[0].EdgeTags.Should().Contain("in");
+        points[1].EdgeTags.Should().Contain("out");
+        points[1].EdgeTags.Should().Contain("out");
```

Comment:
I think this is a typo?
```suggestion
        points[2].EdgeTags.Should().Contain("out");
```

Reply:
yes good catch !